### DISCUSSION TITLE
BUG:The to_crs function can only be used on the Windows platform.

### DISCUSSION THREAD

**Discussion by mht2953658596** [geopandas/geopandas#2955] [Installation]
I need to use the GeoDataFrame..to_crs("EPSG:4326",inplace=True) function to convert my vector layer to the WGS84 coordinate system.

My code was written on Windows, but the actual runtime environment is Linux.

On Windows, I need to set the environment variable 'PROJ_LIB'.

```
PROJ_LIB_PATH = r"D:\anaconda3\Lib\site-packages\rasterio\proj_data"
def judege_platform():
    import platform
    if platform.system() == "Windows":
        os.environ['PROJ_LIB'] = PROJ_LIB_PATH
    elif platform.system() == "Linux":
        pass
```

But I cannot find the location of 'PROJ_LIB' on Linux.

`
result_gdf.to_crs("EPSG:4326",inplace=True)`

However, it will raise an error on Linux.

File "/root/anaconda3/envs/dzpro/lib/python3.8/site-packages/geopandas/geodataframe.py", line 1364, in to_crs
    geom = df.geometry.to_crs(crs=crs, epsg=epsg)
  File "/root/anaconda3/envs/dzpro/lib/python3.8/site-packages/geopandas/geoseries.py", line 1124, in to_crs
    self.values.to_crs(crs=crs, epsg=epsg), index=self.index, name=self.name
  File "/root/anaconda3/envs/dzpro/lib/python3.8/site-packages/geopandas/array.py", line 779, in to_crs
    new_data = vectorized.transform(self.data, transformer.transform)
  File "/root/anaconda3/envs/dzpro/lib/python3.8/site-packages/geopandas/_vectorized.py", line 1114, in transform
    new_coords_z = func(coords_z[:, 0], coords_z[:, 1], coords_z[:, 2])
  File "/root/anaconda3/envs/dzpro/lib/python3.8/site-packages/pyproj/transformer.py", line 430, in transform
    self._transformer._transform(
  File "pyproj/_transformer.pyx", line 459, in pyproj._transformer._Transformer._transform
pyproj.exceptions.ProjError: x, y, z, and time must be same size

SOS！！！！

**Comment by mht2953658596** [comment]
I have also set the 'PROJ_LIB' environment variable on the Linux platform.
```
def judege_platform():
    import platform
    if platform.system() == "Windows":
        os.environ['PROJ_LIB'] = PROJ_LIB_PATH
    elif platform.system() == "Linux":
        os.environ['PROJ_LIB'] ='/root/anaconda3/lib/python3.8/site-packages/rasterio/proj_data'
```


However, this is not useful.

**Comment by mht2953658596** [comment]
gdal                      3.0.2            py38h40f10ac_6    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
geoalchemy2               0.14.0             pyhd8ed1ab_0    conda-forge
geojson                   3.0.1              pyhd8ed1ab_0    conda-forge
geopandas                 0.12.2           py38h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
geopandas-base            0.12.2           py38h06a4308_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
geos                      3.8.0                he6710b0_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
geoserver-rest            2.5.0              pyhd8ed1ab_0    conda-forge
geotiff                   1.7.0                h2a26cda_1    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
giflib                    5.2.1                h5eee18b_3    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
glib                      2.69.1               he621ea3_2    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
sqlalchemy                1.4.39           py38h5eee18b_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
rasterio                  1.2.10           py38h9a6c7ee_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main
rasterstats               0.19.0             pyhd8ed1ab_0    conda-forge
shapely                   2.0.1            py38h0fbb895_0    https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main

**Comment by martinfleis** [comment]
This is a result of incorrectly installed and configured `pyproj` and `PROJ`.

I don't know what channel a `https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main` is but I suspect that its usage over `conda-forge` is causing the issue. Please refer to the pyproj documentation for further troubleshooting once you avoid mixed channels in your conda environment.

Also, ensure that you are using the latest versions of geopandas and pyproj. You haven't reported pyproj but geopandas is outdated.
